robobase/envs/bigym.py

```diff
@@ -1,93 +1,330 @@
-    RescaleFromTanh,
+    RescaleFromTanhWithMinMax,
     OnehotTime,
     ActionSequence,
     AppendDemoInfo,
     FrameStack,
     ConcatDim,
+    RecedingHorizonControl,
 )
 from omegaconf import DictConfig
+from bigym.utils.observation_config import ObservationConfig, CameraConfig
+from bigym.action_modes import PelvisDof
+import multiprocessing as mp
+import logging
+import numpy as np
+
+from demonstrations.demo import DemoStep
+from demonstrations.demo_store import DemoStore
+from demonstrations.utils import Metadata
+
+from typing import List, Dict, Tuple, Callable
+import copy
 
 UNIT_TEST = False
 
 
-class ActionModeType(Enum):
-    TORQUE = "TORQUE"
-    JOINT_POSITION = "JOINT_POSITION"
+def rescale_demo_actions(
+    rescale_fn: Callable, demos: List[List[DemoStep]], cfg: DictConfig
+):
+    """Rescale actions in demonstrations to [-1, 1] Tanh space.
+    This is because RoboBase assumes everything to be in [-1, 1] space.
 
+    Args:
+        rescale_fn: callable that takes info containing demo action and cfg and
+            outputs the rescaled action
+        demos: list of demo episodes whose actions are raw, i.e., not scaled
+        cfg: Configs
 
-def _task_name_to_env_class(task_name: str) -> type[BiGymEnv]:
-    match task_name:
-        case "reach_target":
-            return ReachTarget
-        case "move_plate_between_drainers":
-            return MovePlateBetweenDrainers
-    raise NotImplementedError("Env Not Implemented Yet.")
+    Returns:
+        List[Demo]: list of demo episodes whose actions are rescaled
+    """
+    for demo in demos:
+        for step in demo:
+            info = step.info
+            if "demo_action" in info:
+                # Rescale demo actions
+                info["demo_action"] = rescale_fn(info, cfg)
+    return demos
 
 
-def _create_action_mode(action_mode: str) -> ActionMode:
-    if action_mode == ActionModeType.TORQUE.value:
-        return TorqueActionMode()
-    elif action_mode == ActionModeType.JOINT_POSITION.value:
-        return JointPositionActionMode()
+def _task_name_to_env_class(task_name: str) -> type[BiGymEnv]:
+    return TASK_MAP[task_name]
 
 
 class BiGymEnvFactory(EnvFactory):
-    def _wrap_env(self, env, cfg):
-        env = RescaleFromTanh(env)
-        env = ConcatDim(env, 1, -1, "low_dim_state")
-        env = TimeLimit(env, cfg.env.episode_length)
+    def _wrap_env(self, env, cfg, demo_env=False, train=True, return_raw_spaces=False):
+        # last two are grippers
+        assert cfg.demos > 0
+        assert cfg.action_repeat == 1
+
+        action_space = copy.deepcopy(env.action_space)
+        observation_space = copy.deepcopy(env.observation_space)
+
+        env = RescaleFromTanhWithMinMax(
+            env=env,
+            action_stats=self._action_stats,
+            min_max_margin=cfg.min_max_margin,
+        )
+        obs_stats = None
+        if cfg.norm_obs:
+            obs_stats = self._obs_stats
+
+        env = ConcatDim(
```

Comment:
It is not obvious from the name 'ConcatDim' that it is also normalizing the low dim obs, maybe change the name or leave an explicit comment?

Reply:
Okay makes sense to me. I'll add some comments to explain.